dogma.go

```diff
@@ -104,15 +108,15 @@ func NewClientWithToken(baseURL, token string, transport http.RoundTripper) (*Cl
 // it will throw an error.
 func DefaultOauth2Transport(baseURL, token string, transport http.RoundTripper) (*oauth2.Transport, error) {
 	if len(token) == 0 {
-		return nil, errors.New("token should not be empty")
+		return nil, ErrTokenInvalid
 	}
 	if transport == nil {
-		return nil, errors.New("transport should not be nil")
+		return nil, ErrTransportMustBeSet
 	}
 
 	_, ok := transport.(*oauth2.Transport)
 	if ok {
-		return nil, errors.New("transport cannot be oauth2.Transport")
+		return nil, ErrTransportMustNotBeOauth2
```

Comment:
`OAuth2` that @trustin suggested?
and Could you change the `Oauth2` in dogma.go that I made mistakes as well?